src/HTSPDemuxer.cpp

```diff
@@ -210,12 +210,16 @@ void CHTSPDemuxer::Weight ( enum eSubscriptionWeight weight )
 PVR_ERROR CHTSPDemuxer::CurrentStreams ( PVR_STREAM_PROPERTIES *props )
 {
   CLockObject lock(m_mutex);
-  for (int i = 0; i < m_streams.iStreamCount; i++)
+
+  /* Truncate the list of streams to prevent overflows */
+  m_streams.resize(PVR_STREAM_MAX_STREAMS);
+
+  for (size_t i = 0; i < m_streams.size() && i < PVR_STREAM_MAX_STREAMS; i++)
   {
-    memcpy(&props->stream[i], &m_streams.stream[i], sizeof(PVR_STREAM_PROPERTIES::PVR_STREAM));
+    memcpy(&props->stream[i], &m_streams.at(i), sizeof(PVR_STREAM_PROPERTIES::PVR_STREAM));
   }
 
-  props->iStreamCount = m_streams.iStreamCount;
+  props->iStreamCount = static_cast<unsigned int>(m_streams.size());
```

Comment:
Why the resize? Why do you put more than elements PVR_STREAM_MAX_STREAM into the vector, solely for dropping them later? If you do never add > PVR_STREAM_MAX_STREAMS to the vector, you do not need to resize and you do not need to check anything else than size() here.